src/scrub.cpp

```diff
@@ -0,0 +1,20 @@
+#include <Rcpp.h>
+#include <regex>
+
+using namespace Rcpp;
+// This is 10 times slower than the R equivalent.
```

Comment:
10 times slower? Can you explain this comment? If so, should we just use plain R?

Reply:
The current c++ implementation of scrub is always at least 50% slower than:
```
scrub_r <- function(x) gsub(
  "\u0027\u0027|\u2018\u2018|\u2019\u2019|\u02BC\u02BC|\u0060\u0060|\u00BB|\u201C|\u201D|\u2033|\u3003|\u301E|\u301F|\u0022|\u02BA|\u02DD|\u275E|\u030B|\u030E|\u030F|\u05F4|\u2018|\u2019|\u02BC|\u02C8|\u0060|\u030D|\u02B9|\u00B4|\u0301|\u0374|\u0384|\u2032|\u275C|\u05F3|\u055A|\u055B|\u055D|\u0599|\u059C|\u059D|\u059E|\u00B0|\u02DA|\u030A|\u2070|\u0366|\u05AF|\u00BA|d|g|\u003A",
  "'",
  x
)
```

Probably my c++ implementation is poorly optimised.

Here are the results of a microbrenchmark evaluation:
```
               expr     min      lq     mean   median       uq     max neval
   scrub_r(strings) 10.0289 11.2084 13.16182 12.10685 14.21595 59.7182  1000
 scrub_cpp(strings) 17.1602 18.3274 20.01803 18.98985 20.08735 64.1969  1000
```

Reply:
Thanks for this. I don't know how to make the c++ implementation any faster. Maybe we should stick with your R version `scrub_r` ? What do you think?

Reply:
any thoughts?

Reply:
Well, I wanted to use profiling to make sure that scrub_r would not be a bottleneck of the whole parsing process but even if it was I would still don't know how to speed up scrub_cpp.
So scrub_r is the way I guess. I'll delete scrub_cpp and write scrub_r inside scrub() in zzz.R.

Reply:
thanks